instfixlib.sh

```diff
@@ -0,0 +1,535 @@
+		IOB_USER="iobroker"
+		;;
+	"Darwin")
+		# OSX and Linux are the same in terms of install procedure
+		HOST_PLATFORM="osx"
+		ROOT_GROUP="wheel"
+		INSTALL_CMD="brew"
+		IOB_DIR="/usr/local/iobroker"
+		IOB_USER="$USER"
+		;;
+	"FreeBSD")
+		HOST_PLATFORM="freebsd"
+		ROOT_GROUP="wheel"
+		INSTALL_CMD="pkg"
+		IOB_DIR="/opt/iobroker"
+		IOB_USER="iobroker"
+		;;
+	*)
+		# The following should never happen, but better be safe than sorry
+		echo "Unsupported platform $unamestr"
+		exit 1
+		;;
+	esac
+}
+
+install_package_linux() {
+	package="$1"
+	# Test if the package is installed
+	dpkg -s "$package" &> /dev/null
+	if [ $? -ne 0 ]; then
+		if [ "$INSTALL_CMD" = "yum" ]; then
+			# Install it
+			errormessage=$( $SUDOX yum install -q -y $package > /dev/null 2>&1)
+		else
+			# Install it
+			errormessage=$( $SUDOX $INSTALL_CMD install -yq --no-install-recommends $package > /dev/null 2>&1)
+		fi
+
+		# Hide "Error: Nothing to do"
+		if [ "$errormessage" != "Error: Nothing to do" ]; then
+			if [ "$errormessage" != "" ]; then
+				echo $errormessage
+			fi
+			echo "Installed $package"
+		fi
+	fi
+}
+
+install_package_freebsd() {
+	package="$1"
+	# check if package is installed (pkg is nice enough to provide us with a exitcode)
+	if ! $INSTALL_CMD info "$1" >/dev/null 2>&1; then
+		# Install it
+		$SUDOX $INSTALL_CMD install --yes --quiet "$1" > /dev/null
+		echo "Installed $package"
+	fi
+}
+
+install_package_macos() {
+	package="$1"
+	# Test if the package is installed (Use brew to install essential tools)
+	$INSTALL_CMD list | grep "$package" &> /dev/null
+	if [ $? -ne 0 ]; then
+		# Install it
+		$INSTALL_CMD install $package &> /dev/null
+		if [ $? -eq 0 ]; then
+			echo "Installed $package"
+		else
+			echo "$package was not installed"
+		fi
+	fi
+}
+
+install_package() {
+	case "$HOST_PLATFORM" in
+		"linux")
+			install_package_linux $1
+		;;
+		"osx")
+			install_package_macos $1
+		;;
+		"freebsd")
+			install_package_freebsd $1
+		;;
+		# The following should never happen, but better be safe than sorry
+		*)
+			echo "Unsupported platform $HOST_PLATFORM"
+		;;
+	esac
+}
+
+disable_npm_audit() {
+	# Make sure the npmrc file exists
+	$SUDOX touch .npmrc
+	# If .npmrc does not contain "audit=false", we need to change it
+	$SUDOX grep -q -E "^audit=false" .npmrc &> /dev/null
+	if [ $? -ne 0 ]; then
+		# Remember its contents (minus any possible audit=true)
+		NPMRC_FILE=$($SUDOX grep -v -E "^audit=true" .npmrc)
+		# And write it back
+		write_to_file "$NPMRC_FILE" .npmrc
+		# Append the line to disable audit
+		append_to_file "# disable npm audit warnings" .npmrc
+		append_to_file "audit=false" .npmrc
+	fi
+
+#ADOE: IF (INSTALLER) THEN ...
+	# Make sure that npm can access the .npmrc
+	if [ "$HOST_PLATFORM" = "osx" ]; then
+		$SUDOX chown -R $USER .npmrc
+	else
+		$SUDOX chown -R $USER:$USER .npmrc
+	fi
+#ADOE: ELSE IF (FIXER) THEN ...
+	# No need to change the permissions, since we're doing that soon anyways
+#ADOE: ENDIF
```

Comment:
Are these comments TODOs?

Reply:
If not, how do they work?

Reply:
Its a hint that the INSTALLER and the FIXER differ in this points. And i am not able to decide, which one is correct or if we need both.
If we need both, then the function could:
a) be splitted into two, one for the INSTALL case and one for the FIX case
b) or create a variable e.g. ISFIX=true/false, then: if [ "$ISFIX" = true ]; then ... else ... fi

Reply:
let's do b)